src/core/condense/index.ts

```diff
@@ -90,7 +104,50 @@ export async function summarizeConversation(
 		({ role, content }) => ({ role, content }),
 	)
 	// Note: this doesn't need to be a stream, consider using something like apiHandler.completePrompt
-	const stream = apiHandler.createMessage(SUMMARY_PROMPT, requestMessages)
+	// Use custom prompt if provided and non-empty, otherwise use the default SUMMARY_PROMPT
+	const promptToUse = customCondensingPrompt?.trim() ? customCondensingPrompt.trim() : SUMMARY_PROMPT
+
+	// Use condensing API handler if provided, otherwise use main API handler
+	let handlerToUse = condensingApiHandler || apiHandler
+
+	// Check if the chosen handler supports the required functionality
+	if (!handlerToUse || typeof handlerToUse.createMessage !== "function") {
+		console.warn(
+			"Chosen API handler for condensing does not support message creation or is invalid, falling back to main apiHandler.",
+		)
+		handlerToUse = apiHandler // Fallback to the main, presumably valid, apiHandler
+		// Ensure the main apiHandler itself is valid before this point or add another check.
+		if (!handlerToUse || typeof handlerToUse.createMessage !== "function") {
+			// This case should ideally not happen if main apiHandler is always valid.
+			// Consider throwing an error or returning a specific error response.
+			console.error("Main API handler is also invalid for condensing. Cannot proceed.")
+			// Return an appropriate error structure for SummarizeResponse
+			return {
+				messages,
+				summary: "",
+				cost: 0,
+				newContextTokens: 0,
+			}
+		}
+	}
+
+	// Log when using custom prompt for debugging and telemetry
+	if (customCondensingPrompt?.trim()) {
+		console.log(`Task [${taskId}]: Using custom condensing prompt.`)
+		// TODO: Add telemetry for custom condensing prompt usage
+		// This would require extending the telemetry service with a new method like:
+		// telemetryService.captureCustomCondensingPromptUsed(taskId);
+	}
+
+	// Log when using custom API handler for condensing
+	if (condensingApiHandler) {
+		console.log(`Task [${taskId}]: Using custom API handler for condensing.`)
+		// TODO: Add telemetry for custom condensing API handler usage
+		// This would require extending the telemetry service with a new method like:
+		// telemetryService.captureCustomCondensingApiUsed(taskId, condensingApiConfigId);
+	}
```

Comment:
Instead of this, how about we just add a couple new properties to `	telemetryService.captureContextCondensed(taskId, isAutomaticTrigger ?? false)` above?